### PR TITLE
feat: add OpenResty demo running on docker

### PR DESCRIPTION
**Pull request by laurodd**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,41 @@
-# datadome-openresty-demo
-DataDome OpenResty module demo using docker.
+# DataDome OpenResty demo
+
+This repo tests the DataDome OpenResty module using docker.
```
Comment (Lauredg) [inline README.md]:
```suggestion
This repository provides an example of the DataDome OpenResty module using Docker.
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,41 @@
...
+
+For additional information, please check our [public documentation](https://docs.datadome.co/docs/openresty)
```
Comment (Lauredg) [inline README.md]:
```suggestion
For additional information, please check our [public documentation](https://docs.datadome.co/docs/openresty).
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,41 @@
...
+
+## 1 - Configuration
+
+Open file `nginx.conf` and set your `sever-side-key`
```
Comment (Lauredg) [inline README.md]:
```suggestion
Open the file `nginx.conf` and set your `server-side-key` (found inside your [DataDome Dashboard](https://app.datadome.co/dashboard/management/integrations)).
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,41 @@
...
+docker run -d --rm --name myDataDomeOpenResty -p 8282:80 my-datadome-openresty-image
+```
+
+## 4 - Test
+
+```
+curl -v  http://localhost:8282/
+```
+
+You should be able to see DataDome cookie
```
Comment (Lauredg) [inline README.md]:
```suggestion
Your request was protected (the header `X-DataDome: protected` is present) and is shown in your Dashboard.
```

File: nginx.conf [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    keepalive_timeout  65;
+    #gzip  on;
+
+    # DataDome configuration
+    lua_ssl_verify_depth 3;
+    lua_ssl_trusted_certificate /etc/ssl/certs/ca-certificates.crt;
+
+    resolver 8.8.8.8;
+    init_by_lua_block {
+        require("datadome_openresty").setDatadomeConf( { datadome_server_side_key = "server_side_key", } )
```
Comment (MickaelDatadome) [inline nginx.conf]:
```suggestion
        require("datadome_openresty").setDatadomeConf( { datadome_server_side_key = "datadome_server_side_key", } )
```
Can this be passed as a environment variable ?

File: nginx.conf [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+
+http {
+    include       mime.types;
+    default_type  application/octet-stream;
+
+    sendfile        on;
+    #tcp_nopush     on;
+
+    keepalive_timeout  65;
```
Comment (Lauredg) [inline nginx.conf:16]: Is `65` the default value?

File: nginx.conf [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        server_name  localhost;
+
+         location / {
+            access_by_lua_block {
+                if ngx.req.is_internal() == false then
+                    require("datadome_openresty").validateRequest()
+                end
+            }
+	    
+            proxy_pass http://example.org;
```
Comment (MickaelDatadome) [inline nginx.conf:39]: what is this used for ? 
It is misleading - i'd rather have a section commented out with a clear explanation - or removed
bare minimum config is good

File: README.md (hunk 3 above)
Comment (MickaelDatadome) [inline README.md]:
```suggestion
Open file `nginx.conf` and set your `datadome-sever-side-key`
```

**Review by Lauredg — CHANGES_REQUESTED** [review]
A few changes, otherwise it looks good to me :)

File: README.md (hunk 4 above)
Comment (MickaelDatadome) [inline README.md]:
```suggestion
DataDome cookie is present
Request shows in the DataDome dashboard
```

File: nginx.conf [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    default_type  application/octet-stream;
+
+    sendfile        on;
+    #tcp_nopush     on;
+
+    keepalive_timeout  65;
+    #gzip  on;
+
+    # DataDome configuration
+    lua_ssl_verify_depth 3;
```
Comment (MickaelDatadome) [inline nginx.conf:20]: Is this required or is it default value ?
if it's default it can be omitted

**Review by MickaelDatadome — COMMENTED** [review]
I still want to test it :-)
Good stuff thanks !

File: nginx.conf (hunk 8 above)
Comment (laurodd) [reply nginx.conf:20]: It is required, it cannot be omitted.

File: nginx.conf (hunk 7 above)
Comment (laurodd) [reply nginx.conf:39]: this is used to see a web page (example.org) or else we won't see anything.

File: nginx.conf (hunk 6 above)
Comment (laurodd) [reply nginx.conf:16]: this comes from the nginx conf, we just added datadome config and and proxy_pass to a web page.

File: README.md (hunk 3 above)
Comment (laurodd) [reply README.md]: updating it, thanks!

File: README.md (hunk 4 above)
Comment (laurodd) [reply README.md]: just updated with @Lauredg 's suggestion, thanks

File: nginx.conf (hunk 5 above)
Comment (laurodd) [reply nginx.conf]: commited the change.
The idea was to simplify the Dockerfile and have something closer to what the clients do (change this file manually in PROD).

File: README.md (hunk 2 above)
Comment (MickaelDatadome) [reply README.md]: picky... (@Lauredg )

File: nginx.conf (hunk 7 above)
Comment (MickaelDatadome) [reply nginx.conf:39]: shouldn't it be https://point.datashield.co in this case ? :-P

**Review by MickaelDatadome — APPROVED** [review]
Tested on my side, it's working ! ;-)

**Review by Lauredg — APPROVED** [review]

File: nginx.conf [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    lua_ssl_trusted_certificate /etc/ssl/certs/ca-certificates.crt;
+
+    resolver 8.8.8.8;
+    init_by_lua_block {
+        require("datadome_openresty").setDatadomeConf( { datadome_server_side_key = "datadome_server_side_key", } )
```
Comment (Lauredg) [inline nginx.conf:25]: I am being picky but the fact that the two strings were different ( `datadome_server_side_key` and `server_side_key`) before made it easier to understand and search for what should be changed.
In this case we could change the instructions in the README.md to

> Open the file `nginx.conf` and replace `server-side-key` with the value of the DataDome server-side key that you can find in your ...

Comment (Lauredg) [reply nginx.conf:25]: Too late haha

**Pull request merged**